delta/plugins/search/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/search/Search.scala

```diff
@@ -19,9 +28,63 @@ trait Search {
 
 object Search {
 
-  //TODO: modify
-  final def apply: Search = new Search {
-    override def query(payload: JsonObject, qp: Uri.Query)(implicit caller: Caller): IO[SearchRejection, Json] =
-      IO.pure(Json.obj("test" -> true.asJson))
+  final case class TargetProjection(projection: ElasticSearchProjection, view: CompositeView, rev: Long)
+
+  private[search] type ListProjections = () => UIO[Seq[TargetProjection]]
+
+  /**
+    * Constructs a new [[Search]] instance.
+    */
+  final def apply(
+      compositeViews: CompositeViews,
+      acls: Acls,
+      client: ElasticSearchClient,
+      indexingConfig: ExternalIndexingConfig
+  ): Search = {
+
+    val listProjections: ListProjections = () =>
+      compositeViews
+        .list(
+          Pagination.OnePage,
+          CompositeViewSearchParams(deprecated = Some(false), filter = _.id == defaultViewId),
+          Ordering.by(_.createdAt)
+        )
+        .map(
+          _.results
+            .flatMap { entry =>
+              val res = entry.source
+              for {
+                projection   <- res.value.projections.value.find(_.id == defaultProjectionId)
+                esProjection <- projection.asElasticSearch
+              } yield TargetProjection(esProjection, res.value, res.rev)
+            }
+        )
+    apply(listProjections, acls, client, indexingConfig)
   }
+
+  /**
+    * Constructs a new [[Search]] instance.
+    */
+  final def apply(
+      listProjections: ListProjections,
+      acls: Acls,
+      client: ElasticSearchClient,
+      indexingConfig: ExternalIndexingConfig
+  ): Search =
+    new Search {
+
+      override def query(payload: JsonObject, qp: Uri.Query)(implicit caller: Caller): IO[SearchRejection, Json] = {
+        for {
+          allProjections    <- listProjections()
+          accessible        <-
+            acls.authorizeForAny(allProjections.map(v => ProjectAcl(v.view.project) -> v.projection.permission))
+          accessibleProjects = accessible.collect { case (p: ProjectAcl, true) => ProjectRef(p.org, p.project) }.toSet
+          accessibleIndices  = allProjections.collect {
+                                 case v if accessibleProjects.contains(v.view.project) =>
+                                   CompositeViews.index(v.projection, v.view, v.rev, indexingConfig.prefix).value
+                               }
+          results           <- client.search(payload, accessibleIndices.toSet, qp)().mapError(WrappedElasticSearchClientError)
```

Comment:
There is a problem if there are a lot of indices as the indices are part of the url (same constraint as AggregatedElasticSearchViews)
Also with this approach we will hit all indices accessible to the user even if we want to search on a subset of them (we could handle this problem later though).
And if someone wants to search on an index he does not have access to in his query, he will have an empty result and not a forbidden response (but it may be ok for now)

Maybe if we create a ES alias on the projections at creation with an alias looking like `{prefix}_search_{org}_{proj}` it would also be easier to filter them as we would have to loop on composite views but just fetch the user acls to build the accessible indices ?

Then one way to go (even it would hit all search indices) would be to make a query like:
```
GET "/{prefix}_search_{org}_{proj}/_search"
{
  "query": {
    "terms": {
      "_index": [
          "{prefix}_search_{org}_{proj}",
          "{prefix}_search_{org}_{proj}",
          ... //all indices] 
    }
    }
}
```
that would have to be merged with a ES bool query with the one passed by fusion

Reply:
There's a lot of room to improve it here, but I'd say is better to address them in a subsequent PR

Reply:
I'll address it in the subsequent PR